lib/formatters/json.js

```diff
@@ -8,30 +6,14 @@
  * JSON formatter.
  * @public
  * @function formatJSON
- * @param    {Object} req  the request object
+ * @param    {Object} req  the request object (not used)
  * @param    {Object} res  the response object
  * @param    {Object} body response body
  * @param    {Function} cb cb
  * @returns  {String}
  */
 function formatJSON(req, res, body, cb) {
-    if (body instanceof Error) {
-        // snoop for RestError or HttpError, but don't rely on
-        // instanceof
-        res.statusCode = body.statusCode || 500;
-
-        if (body.body) {
-            body = body.body;
-        } else {
-            body = {
-                message: body.message
-            };
-        }
-    } else if (Buffer.isBuffer(body)) {
-        body = body.toString('base64');
-    }
-
-    var data = JSON.stringify(body);
+    var data = body ? JSON.stringify(body) : '';
```

Comment:
I thought about that. A null or undefined body represents a body payload of nothing. What you suggest is 2 bytes where none before were. If the accept type is application/json but the function returns nothing at all, then really, the accept type doesn't matter. The way I look at it, this code should produce the same results in both directions: f(g(x)) = g(f(x))
```
var thing = JSON.parse(JSON.stringify(null));
console.log(thing); // produces null, good, so this is our f(g(x)) = g(f(x))

var thing = JSON.parse(JSON.stringify(''));
console.log(thing); // produces "\"\""   which isn't right
```
Upon testing my own ideas I don't like my code either. How about in this condition we set the data to null?